### ISSUE TITLE
ConcurrentModificationException on com.devicecollector.CollectorProcess.doInBackground

### ISSUE THREAD

**Issue by zinuzoid** [braintree/braintree_android#53]
I use getDeviceData function to collect data and sent to Braintree
```
mBraintreeDeviceData = Braintree.getDeviceData(this);
```
We got a lot of crash on this ~0.5% of sessions.
Not sure if this related to Braintree but It's cause by library that you included in SDK.

Crash log:
```
java.lang.RuntimeException: An error occured while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:300)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:848)
Caused by: java.util.ConcurrentModificationException
       at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:573)
       at com.devicecollector.CollectorProcess.doInBackground(CollectorProcess.java:102)
       at com.devicecollector.CollectorProcess.doInBackground(CollectorProcess.java:34)
       at android.os.AsyncTask$2.call(AsyncTask.java:288)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:848)
```

**Comment by lkorth** [comment]
@zinuzoid thanks for reporting this crash. We are currently working with our partner that provides this library to get an update to fix the crash. We'll keep you updated.

**Comment by lkorth** [comment]
@zinuzoid I just released `1.7.5` with an updated version of the device collector that should fix this. Please let us know if you are still experiencing crashes.
